From 61a807f7857d0fa754fbae5a7172dee570d90b91 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Tue, 5 Mar 2019 09:11:10 +0100
Subject: [PATCH] [DNM] Squash of lineage-16.0-android-9.0.0_r34

Squashed commit of the following:

commit db3dd8187ce35f26ab0542718f585833975b937f
Merge: 321c5fc 9393ade
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Wed Jan 16 18:57:47 2019 +0000

    Merge cherrypicks of [6072697, 6072075, 6072758, 6072124, 6072885, 6072886, 6072887, 6072580, 6072581, 6072582, 6072583, 6072584, 6072132, 6072195, 6072133, 6072077, 6072134, 6072078, 6072211, 6072762, 6072763, 6072908, 6072909, 6072910, 6072911, 6072912, 6072913, 6072914, 6072930, 6072212, 6072743] into pi-qpr2-release

    Change-Id: I5eb8df5ce122479b918e17d485a22fb671614c67

commit 9393ade3a1dadbbff1293d8dca420a6d155695ec
Author: Steven Moreland <smoreland@google.com>
Date:   Mon Jan 7 14:12:14 2019 -0800

    Fillout requesting SID.

    Bug: 121035042
    Test: boot and check for SIDs coming into hwservicemanager instances
        which were retrieved w/ IServiceManager::getService

    Change-Id: I3a7f6c0b3cc8bab24e0639a3e7121ef536b2603b
    Merged-In: I3a7f6c0b3cc8bab24e0639a3e7121ef536b2603b
    (cherry picked from commit ca8cf63c53b242e4dda5b1da6deaa370470fca1c)

Change-Id: I44ff6bc5639731ef2a68edf51e0e61abec551c83
---
 generateCpp.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/generateCpp.cpp b/generateCpp.cpp
index 4fd3956..fa2a3ca 100644
--- a/generateCpp.cpp
+++ b/generateCpp.cpp
@@ -1238,6 +1238,7 @@ void AST::generateStubSource(Formatter& out, const Interface* iface) const {
         << "_hidl_impl, {SCHED_NORMAL, 0});\n";
     out << "mSchedPolicy = prio.sched_policy;\n";
     out << "mSchedPriority = prio.prio;\n";
+    out << "setRequestingSid(::android::hardware::details::gServiceSidMap.get(_hidl_impl, false));\n";
     out.unindent();
 
     out.unindent();
-- 
2.17.1

